### DISCUSSION TITLE
GitHub Mobile 2FA disappears, is not re-triggerable

### DISCUSSION THREAD

**Discussion by Grantimus9** [community/community#33741] [Mobile]
I frequently request 2FA via GitHub Mobile, and encounter the following errors preventing me from continuing: 

1. Clicking mobile notification asking me to submit code results in "no requests found". Did the notification arrive without a request? How do I refresh or re-retrieve? 

2. Frequently I do not receive a push notification unless the app is already open. If I open the app seconds after requesting a 2FA via GH Mobile, there is no notification or place to confirm my login attempt. Please provide a means to surface a path to confirming a 2FA once the app is open but notification has been dismissed.

**Comment by eliperkins** [comment]
Hey @Grantimus9 ! That does sound like some odd behavior related to GitHub Mobile 2FA. Can I ask a few questions to gather some more info to help our engineering team debug?

- What device and operating system are you on? (e.g., iPhone 12 + iOS 15, Pixel 6a + Android 13)
- Do you have multiple devices with GitHub Mobile installed on them?
- Have you uninstalled and reinstalled the app, or logged in and out of the app several times lately?

> Please provide a means to surface a path to confirming a 2FA once the app is open but notification has been dismissed

This is a great point. The app should automatically refresh 2FA requests upon opening the app, but making this behavior more explicit with some UI would be helpful for sure. I'll follow up with the team on this one.

Looking forward to troubleshooting this for ya!

**Reply by Grantimus9** [reply]
Pixel 5, Android 11 for all instances. 

Only this one device. 

Haven't logged out/in or uninstalled the app either. 

Btw I'm not sure "refresh" the request is what users want; we just need the ability to respond to an open 2FA challenge; some way to reveal the input form to type in the code presented to me on my browser.

**Reply by Grantimus9** [reply]
Still an active bug, I'm now on Android 12 FWIW. To reiterate: a way to surface the ability to respond to a 2FA challenge needs to be in the app _other than the easily missed notification_. Add something like "2FA Requests" under profile or settings and have the code submission form exist there?

**Reply by 4wk-** [reply]
Pixel 5 Android 14. Most of the time (> 75%), taping the android notification display "no verification requests found", sadly.
It's really a shame that in 2024 still we don't have anywhere on the app a way to see "current 2FA requests"... 
This has been an issue since 2022...

**Reply by asyncjake** [reply]
@eliperkins pixel 4a, same problem

**Comment by heydarli** [comment]
I'm having the same issue on Pixel 6a + Android 13. After clicking the notification the app shows "no requests found" and there is no way to recover from this.
- I don't have multiple devices  and 
- have not logged in/out recently. 

It works sometimes though.

**Reply by CullenStClair** [reply]
Likewise on Pixel 6 + Android 13. The verification request appears properly if the GitHub Mobile app is open in the foreground when the request is sent, but otherwise the above issue occurs.

**Reply by 199970** [reply]
Cool

**Comment by matheusmoreira** [comment]
Samsung Galaxy Note 9. Same issue: I get the notification but the app says "no verification requests found" when I tap it. This is my only mobile device and I have not logged out of or into the app recently.

**Reply by 199970** [reply]
Good

**Comment by drguildo** [comment]
This is clearly a bug, so it should be reported as such. This then raises the question why is this a closed source project, and why isn't there a proper issue tracker?

My "fix" is to stop using GitHub Mobile for authentication. Switch to a proper 2FA app that actually works.

**Reply by bomba466** [reply]
I feel this is certificates

**Reply by 199970** [reply]
Cool

**Comment by kxcrl** [comment]
Just coming here to say that I'm having the same issue.

- Began in January of this year.
- Android 12.
- No other devices connected to github.
- Intermittent for a couple of days. Now fails 100% of the time with "No verification requests found" after sending a push notification.
- Resetting, reinstalling, clearing data, etc. made no difference.

**Comment by luke-quest** [comment]
This happened to me around the 1/5/2024. 100% 2FA with Github mobile didn't work. Deleted storage and uninstalled the app and then I could access. We have an org with SAML auth, which feels like it might be connected.

**Reply by bomba466** [reply]
someone transferred files from mobile and PC

**Comment by tomwhoiscontrary** [comment]
This happens to me too. Currently on a Pixel 7a running Android 14. Happened on my previous phone too, a Pixel 4a i think. AFAICR, it has happened consistently ever since i had to start using 2FA. I only use the GitHub app on this phone, and i have not reinstalled or logged in or out recently. I am an ordinary personal user, so no corporate SSO involved.

**Comment by Jerboa-app** [comment]
This happens almost every time for me, and did just now, I get notifications for "sudo mode" and then a popup "no requests found" once I open the Github app or click on the notification. It seems to be more common when I need to enter a two digit code (unless I miss-remember sometimes it is just a click?). 

Only installed on one device Sony Xperia i (J8110, Android 11 build 55.2.A.4332) has been happening *for years* since I first started using Github mobile. Ordinary user as well.

I notice the 2-digit code format is the same as Microsoft authenticator, maybe there are some crossed wires... I do have that installed. No idea how this is implemented in github's app but if this is a downstream codebase maybe it is related.

**Comment by dJani97** [comment]
This "no verification requests found" happens to me 95% of the time I try to use the app for 2FA. I have already given up on it and only rely on SMS codes.

I'm on Samsung's Android 14.

**Comment by kotanin** [comment]
Pixel 7, Android 14
same issue

**Comment by Vectorial1024** [comment]
I have a similar issue, but it seems that if I first open the GitHub mobile app, and then try to login via web, then the authentication popup will correctly ask me for the verification code instead of telling me "no verification requests found".

The many times I got "no verification requests found" were all me clicking the push notification without having the app already running, i.e. the app is running for the first time since boot when I try to do 2FA. I suspect this is because the app genuinely did not see any verification request because the app was not started and therefore did not receive anything.

**Reply by MarkusSecundus** [reply]
Same issue. Installed on two phones - Xperia 5 II (Android 10, bld. 58.0.A.12.23) and XZ Premium (Android 9, bld. 47.2.A.10.107)

**Reply by Jerboa-app** [reply]
> it seems that if I first open the GitHub mobile app, and then try to login via web, then the authentication popup will correctly ask me for the verification code instead of telling me "no verification requests found"

Just tried that and it seems to work, will at least save some time.

**Comment by KinNeko-De** [comment]
I have had this issue multiple times, but retry always works, mostly after the first retry.

Rarely, I do not get a notification. Happend only few times.

Most of the time I get a notification. When I click on it, Github Mobiles opens and the "no request was found" error.

Samsung Android 14

Only one device with Gituhn Mobile is installed. 
but I have also Microsoft Authenticator installed which is configured as a 2FA method.
Sometimes this authenticator is used instead of my preferred 2FA.

I will now deactivate Microsoft Authenticator. Maybe @Jerboa-app guessed right :)

**Reply by KinNeko-De** [reply]
Worked until now

**Reply by KinNeko-De** [reply]
> Worked until now

Happennd to me again. Getting worse in the last month

**Comment by Lyra-Ravenlock** [comment]
I've been having the same issue on a Motorola g14. Only one device

**Comment by ca-d** [comment]
Had the same issue on my Pixel 4, got a new Oneplus Nord 4 now, the issue still persists.

**Comment by joemar25** [comment]
Hi @Grantimus9, I understand your concern about the GitHub Mobile 2FA issue. It seems like you're experiencing a problem where the 2FA request disappears,and you're unable to retrieve it.
From what I've gathered, it looks like this is a bug in the GitHub Mobile app. Many users have reported similar issues, including not receiving push notifications unless the app is already open, and not being able to find a place to confirm their login attempt after opening the app. 
One possible solution is to add a "2FA Requests" section in the app where users can find and respond to open 2FA challenges. In the meantime, you can try a workaround: open the GitHub mobile app before attempting to log in via the web. This might help you receive the authentication popup correctly.

**Comment by kimhanse** [comment]
![image](https://github.com/user-attachments/assets/36c7804c-e003-4ebb-bbc0-c8a1e93da915)

On most logins I get the "No verification requests found" and end up using TOTP codes instead.

**Reply by Vectorial1024** [reply]
Try opening the app before triggering the 2FA, but I will admit this does not seem to occur to me that much recently

**Comment by ghost** [comment]
yup, the github mobile app is the most unreliable 2fa. I also get the no requests found error most of the time. It works sometimes though. Please create an issue and don't paste those "I understand you have a problem" answers. We aren't idiots :).

Have a nice day.

**Comment by domdfcoding** [comment]
Still happening in mid December 2024

**Comment by jureyherrera** [comment]
I have the same issue with a Fariphone 4 (Android version 13), but it works if I first open the app before launching the 2FA, as @Vectorial1024 suggested.

**Reply by Vectorial1024** [reply]
A link to the mentioned workaround https://github.com/orgs/community/discussions/33741#discussioncomment-9890182

**Comment by douglasg14b** [comment]
Still happens January 2025....

Intermittently. Extremely frusterating

**Comment by ksheku97** [comment]
Start work

**Comment by 199970** [comment]
Cool

**Comment by james-wilson-ln** [comment]
Still happens at the end of Feb 2025.

**Comment by JCrabtree911** [comment]
March of 2025 and this hasn't been fixed.

**Reply by Vectorial1024** [reply]
A link to the mentioned workaround https://github.com/orgs/community/discussions/33741#discussioncomment-9890182

**Comment by Stuie** [comment]
This has been bugging me for literal years now. All it does is slow down what should be a nearly friction-less interaction.

1. Enter my username and password from my password manager
2. Tap the notification on my phone
3. It fails 99% of the time
4. Choose one of the 3 or 4 actually working 2FA options
5. Finally get logged in and try to remember what I was going to GitHub for

It's so frustrating that this has been an issue for so many people for so long. Are there people out there for whom this works every time?
